lib/tapioca/generator.rb

```diff
@@ -692,5 +694,13 @@ def report_diff_and_exit_if_out_of_date(diff, command)
         exit(1)
       end
     end
+
+    sig { void }
+    def abort_if_pending_migrations!
+      return unless defined?(Rails)
+
+      Rails.application.load_tasks
```

Comment:
What if `Rails` is defined but we're not in an application? 

Should we protect this behind a `return unless File.exist?("config/application.rb")` like its done for [`load_rails`](https://github.com/Shopify/tapioca/blob/ccd7c75305f4c3c5d0ad7f10d1bd63544f14e565/lib/tapioca/loader.rb#L31)?

Reply:
Yeah. For example, that is exactly the case in our support repo. `Rails` is defined by not in the way you think. So the added test really fails in this PR.

Does eager loading the Rails application not load tasks, already? Do we really need this line?

Reply:
I tested with an application, rake tasks are not loaded during eager load. Probably because the tasks as typically invoked from a CLI context and not used from inside the application running. Without that line the conditional below always fails because the task is never defined.